### PR TITLE
Add second lecture draft

### PR DESCRIPTION
**Pull request by joelostblom**
I included all the output in the notebook so that it is easy to view and comment here.

### CODE REVIEW

**Comment by joelostblom** [comment]
@OrdiNeu I updated this notebook, mostly reordering the intro to Python part. I believe this might be a bit short for 2 h 30 min, so I am considering moving ~20-30 min from the next lecture here, I'll send everyone an email regarding this. But please read through the notebook and let me know what you think. in regards to the time estimations.

File: code/2-data-analysis-in-python.ipynb [hunk 1]
```diff
@@ -0,0 +1,5073 @@
...
+   "metadata": {},
+   "source": [
+    "The output from `type()` is formatted slightly differently when it is printed."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Python also allows to use the comparison and logic operators `<, >, ==, !=, <=, >=` and statements of identity such as `and, or, not`, which will return either `True` or `False`."
```
Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: statements of identity?

File: code/2-data-analysis-in-python.ipynb [hunk 2]
```diff
@@ -0,0 +1,5073 @@
...
+   ],
+   "source": [
+    "bad = {[1, 2, 3]: 3}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The message within the red background is an error message referred to as a traceback. This message pinpoints what line in the code cell resulted in an error when it was executed, by pointing at it with an arrow (`---->`). This is helpful in figuring out what went wrong.\n",
```
Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: There's no red background here, for some reason.

File: code/2-data-analysis-in-python.ipynb [hunk 3]
```diff
@@ -0,0 +1,5073 @@
...
+    "\n",
+    "### Packages \n",
+    "\n",
+    "Since there are so many esoteric tools and functions available in Python, it is unnecessary to include all of them with the basics that are loaded by default when you start the programming language (it would be as if your new phone came with every single app preinstalled). Instead, more advanced functionality is grouped into separate packages, which can be accessed by typing import <package_name> in Python. You can think of this as that you are telling the program which menu items you want to use (similar to how Excel hides the Developer menu by default since most people rarely use it and you need activate it in the settings if you want to access its functionality). Some packages needs to be downloaded before they can be used, just like downloading an addon to a browser or mobile phone.\n",
+    "\n",
+    "Just like in spreadsheet software menus, there are lots of different tools within each Python package. For example, if I want to use numerical Python functions, I can import the numerical python module, numpy. I can then access any function by writing numpy.<function_name>."
```
Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: Link `numpy` like `pandas` below?

File: code/2-data-analysis-in-python.ipynb [hunk 4]
```diff
@@ -0,0 +1,5073 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> ## Challenge - Can you do reassignment in a dictionary?\n",
+    ">\n",
+    "> 1. First check what `rev` is right now (remember `rev` is the name of our dictionary).\n",
+    ">\n",
+    ">    Type:\n",
+    "> ```python\n",
+    "> >>> rev\n",
```
Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: The `>>>` might be a bit confusing, being in the notebook.

File: code/2-data-analysis-in-python.ipynb [hunk 5]
```diff
@@ -0,0 +1,5073 @@
...
+    "This study includes observations of plants, ants, and rodents from 1977 - 2002,\n",
+    "and has been used in over 100 publications. More information is available in\n",
+    "[the abstract of this paper from 2009](\n",
+    "http://onlinelibrary.wiley.com/doi/10.1890/08-1222.1/full). There are several\n",
+    "datasets available related to this study, and we will be working with datasets\n",
+    "that have been preprocessed by the [Data\n",
+    "Carpentry](https://www.datacarpentry.org) to facilitate teaching. These are made\n",
+    "available online as *The Portal Project Teaching Database*, both at the [Data\n",
+    "Carpentry website](http://www.datacarpentry.org/ecology-workshop/data/), and on\n",
+    "[Figshare](https://figshare.com/articles/Portal_Project_Teaching_Database/1314459/6).\n",
```
Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb:1055]: The figshare link is to an old revision; you may want to re-run everything with the latest.

File: code/2-data-analysis-in-python.ipynb [hunk 6]
```diff
@@ -0,0 +1,5073 @@
...
+       "16          1453     11    5  1978        2         NL    M              NaN   \n",
+       "17          1756      4   29  1979        2         NL    M             33.0   \n",
+       "18          1818      5   30  1979        2         NL    M             32.0   \n",
+       "19          1882      7    4  1979        2         NL    M             32.0   \n",
+       "20          2133     10   25  1979        2         NL    F             33.0   \n",
+       "21          2184     11   17  1979        2         NL    F             30.0   \n",
+       "22          2406      1   16  1980        2         NL    F             33.0   \n",
+       "23          2728      3    9  1980        2         NL    F              NaN   \n",
+       "24          3000      5   18  1980        2         NL    F             31.0   \n",
+       "25
```
Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: `DataFrame`

Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: constitute
`DataFrame`

Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: s/but and/but an/

Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: `DataFrame`

Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: There is no `nrow()`.

Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: Probably can drop this one.

Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: running off

Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: No pipes...

Comment (QuLogic) [inline code/2-data-analysis-in-python.ipynb]: Needs updating.

**Review by QuLogic — COMMENTED** [review]
Some challenges require updates.

File: code/2-data-analysis-in-python.ipynb (hunk 5 above)
Comment (joelostblom) [reply code/2-data-analysis-in-python.ipynb:1055]: Hmmm, I rather keep it as is with the old one as long as the students also download the old data.

**Comment by joelostblom** [comment]
Thanks @QuLogic ! Updated based on your suggestions. Challenges are now ready for tomorrow.

**Comment by joelostblom** [comment]
@OrdiNeu This should be good to go now, let me know if you have any questions for tomorrow

**Pull request merged**